model/patch_lifecycle.go

```diff
@@ -438,9 +438,61 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		return nil, errors.WithStack(err)
 	}
 
+	if p.IsParent() {
+		//finalize child patches that have a parentStatus = ""
+		for _, childPatch := range p.Triggers.ChildPatches {
+			err = finalizeChildPatch(ctx, childPatch, requester, githubOauthToken)
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":       "unable to finalize child patch",
+				"childPatch_id": childPatch,
+				"patch_id":      p.Id,
+				"variants":      p.BuildVariants,
+				"tasks":         p.Tasks,
+				"variant_tasks": p.VariantsTasks,
+				"alias":         p.Alias,
+			}))
+			return nil, err
+		}
+	}
+
 	return patchVersion, nil
 }
 
+func finalizeChildPatch(ctx context.Context, childPatchId, requester, githubOauthToken string) error {
+	// if the parentStatus is "", finalize without waiting for the parent patch to finish running
+	childPatchDoc, err := patch.FindOneId(childPatchId)
+	if err != nil {
+		return errors.Errorf("error fetching child patch: %s", err)
+	}
+	if childPatchDoc == nil {
+		return errors.New("child patch not found")
+	}
+
+	intent, err := patch.FindIntent(childPatchId, patch.TriggerIntentType)
+	if err != nil {
+		return errors.Errorf("error fetching child patch intent: %s", err)
+	}
+	if intent == nil {
+		return errors.New("childpatch intent not found")
+	}
+
+	if intent.GetParentStatus() == "" {
```

Comment:
I  think an alternative to adding an getter to the interface is to type assert it to a *TriggerIntent so we'll have access to its fields directly.
We do something similar [here](https://github.com/evergreen-ci/evergreen/blob/a23d98cf3a148308d331931abcc1a4f088a46e0c/units/patch_intent.go#L723-L726)

This would avoid adding a method to the interface which only matters for this one implementation.

Reply:
good idea, thanks!